examples/component_sharing.yaml

```diff
@@ -8,27 +8,28 @@
 # - reference-sharing: sharing reference to a complex object.
 #                   this means the component is only created once, which also leads to DyNet parameters being shared.
 #                   Since YAML doesn't distinguish between a copy and a reference, we implement reference-sharing
-#                   by adding a unique __xnmt_id field to the component that is to be shared. During deserialization,
-#                   whenever a component's __xnmt_id is equal to the corresponding field of a component that has
+#                   by adding a unique _xnmt_id field to the component that is to be shared. During deserialization,
+#                   whenever a component's _xnmt_id is equal to the corresponding field of a component that has
 #                   has already been initialized, we re-use that initialized component instead of creating a new one.
 
-defaults:
+exp1.pretrain: &defaults
```

Comment:
Should this be `&defaults` or `&exp1.pretrain`?
It might be worth thinking about what the canonical syntax here should be, as others will probably use it in the future, and I think perhaps giving the experiment the same name might be better?

Reply:
Sure, that's a good idea, I'll try that!